core/src/main/scala/org/apache/spark/sql/rapids/tool/util/plangraph/PlanGraphTransformer.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.rapids.tool.util.plangraph
+
+import com.nvidia.spark.rapids.tool.planparser.DatabricksParseHelper
+
+import org.apache.spark.sql.execution.ui.{SparkPlanGraphCluster, SparkPlanGraphNode}
+
+
+/**
+ * Object responsible for transforming instances of SparkPlanGraphNode and SparkPlanGraphCluster
+ * to specialized instances (e.g. PhotonSparkPlanGraphNode) if they are of a specialized type.
+ *
+ * This can be extended to handle other types of nodes or clusters (e.g. GPU nodes or Velox nodes).
+ */
+object PlanGraphTransformer {
```

Comment:
Nit: if we have an objectTransformer we could also move the processPlanInfo() to be dynamically decided based on the App type.
But I don't think we need this kind of refactor immediately. We can do that always later if we find it has advantage in complexity or performance

Reply:
Thanks @amahussein for this idea. Yes, we could refactor this to use a class based approach instead. In future, when we add support for other node types (Velox etc), we could refactor this.